examples/contrib/oed/models/bayes_linear.py

```diff
@@ -67,47 +191,76 @@ def bayesian_linear_model(design, w_means=None, w_sqrtlambdas=None, obs_sd=None,
     # response will be shape batch x n
     obs_sd = obs_sd.expand(tau_shape).unsqueeze(-1)
 
-    # Allow different names for different coefficient groups
+    # Build the regression coefficient
     w = []
-    if w_sqrtlambdas is not None:
-        for name, w_sqrtlambda in w_sqrtlambdas.items():
-            w_mean = w_means[name]
-            # Place a normal prior on the regression coefficient
-            w_prior = dist.Normal(w_mean, obs_sd / w_sqrtlambda).independent(1)
-            w.append(pyro.sample(name, w_prior).unsqueeze(-1))
-    w = torch.cat(w, dim=-2)
+    # Allow different names for different coefficient groups
+    # Process fixed effects
+    for name, w_sqrtlambda in w_sqrtlambdas.items():
+        w_mean = w_means[name]
+        # Place a normal prior on the regression coefficient
+        w_prior = dist.Normal(w_mean, obs_sd / w_sqrtlambda).independent(1)
+        w.append(pyro.sample(name, w_prior))
+    # Process random effects
+    for name, group_size in re_group_sizes.items():
+        # Sample `G` once for this group
+        alpha, beta = re_alphas[name], re_betas[name]
+        group_p = alpha.shape[-1]
+        G_prior = dist.Gamma(alpha.expand(tau_shape + (group_p,)),
+                             beta.expand(tau_shape + (group_p,)))
+        G = 1./torch.sqrt(pyro.sample("G_" + name, G_prior))
+        # Repeat `G` for each group
+        repeat_shape = tuple(1 for _ in tau_shape) + (group_size,)
+        u_prior = dist.Normal(torch.tensor(0.), G.repeat(repeat_shape)).independent(1)
+        w.append(pyro.sample(name, u_prior))
+    # Regression coefficient `w` is batch x p
+    w = torch.cat(w, dim=-1)
 
     # Run the regressor forward conditioned on inputs
-    prediction_mean = torch.matmul(design, w).squeeze(-1)
+    prediction_mean = rmv(design, w)
     if response == "normal":
         # y is an n-vector: hence use .independent(1)
         return pyro.sample(response_label, dist.Normal(prediction_mean, obs_sd).independent(1))
     elif response == "bernoulli":
         return pyro.sample(response_label, dist.Bernoulli(logits=prediction_mean).independent(1))
+    elif response == "sigmoid":
+        base_dist = dist.Normal(prediction_mean, obs_sd).independent(1)
+        # You can add loc via the linear model itself
+        k = k.expand(prediction_mean.shape)
+        transforms = [AffineTransform(loc=torch.tensor(0.), scale=k), SigmoidTransform()]
+        response_dist = dist.TransformedDistribution(base_dist, transforms)
+        return pyro.sample(response_label, response_dist)
     else:
         raise ValueError("Unknown response distribution: '{}'".format(response))
 
 
-def normal_inv_gamma_guide(design, obs_sd, w_sizes):
+# TODO replace this guide with one allowing correlation between
+# regression coefficients (necessary!)
+def normal_inv_gamma_family_guide(design, obs_sd, w_sizes, mf=False):
```

Comment:
This is flagged for rewriting because it ignores all correlations between coefficients